frontend/src/views/mspcontracts/UpdateContractAddress.vue

```diff
@@ -0,0 +1,218 @@
+
+  setup() {
+    return {
+      v$: useVuelidate(),
+    }
+  },
+  data() {
+    return {
+      submitting: false,
+      updateOk: false,
+      updateMode: true,
+      //Update Contract Address Fields
+      phn: '',
+      groupNumber: '',
+      homeAddress: {
+        addressLine1: '',
+        addressLine2: '',
+        addressLine3: '',
+        addressLine4: '',
+        postalCode: '',
+      },
+      mailingAddress: {
+        addressLine1: '',
+        addressLine2: '',
+        addressLine3: '',
+        addressLine4: '',
+        postalCode: '',
+      },
+      result: {
+        phn: '',
+        status: '',
+        message: '',
+      },
+    }
+  },
+
+  methods: {
+    async submitForm() {
+      this.submitting = true
+      this.updateOk = false
+      this.updateMode = true
+      this.$store.commit('alert/dismissAlert')
+      try {
+        const isValid = await this.v$.$validate()
+        if (!isValid) {
+          this.$store.commit('alert/setErrorAlert')
+          this.submitting = false
+          return
+        }
+        this.result = (
+          await MspContractsService.updateContractAddress({
+            phn: this.phn,
+            groupNumber: this.groupNumber,
+            homeAddress: {
+              addressLine1: this.homeAddress.addressLine1,
+              addressLine2: this.homeAddress.addressLine2,
+              addressLine3: this.homeAddress.addressLine3,
+              addressLine4: this.homeAddress.addressLine4,
+              postalCode: this.homeAddress.postalCode,
+            },
+            mailingAddress: {
+              addressLine1: this.mailingAddress.addressLine1,
+              addressLine2: this.mailingAddress.addressLine2,
+              addressLine3: this.mailingAddress.addressLine3,
+              addressLine4: this.mailingAddress.addressLine4,
+              postalCode: this.mailingAddress.postalCode,
+            },
+          })
+        ).data
+
+        if (this.result.status === 'error') {
+          this.$store.commit('alert/setErrorAlert', this.result.message)
+          return
+        }
+
+        if (this.result?.status === 'success') {
+          this.updateMode = false
+          this.updateOk = true
+          this.$store.commit('alert/setSuccessAlert', this.result.message)
+          return
+        }
+      } catch (err) {
+        this.$store.commit('alert/setErrorAlert', `${err}`)
+      } finally {
+        this.submitting = false
+      }
+    },
+    resetForm() {
+      this.groupNumber = ''
+      this.phn = ''
+      this.homeAddress.addressLine1 = ''
+      this.homeAddress.addressLine2 = ''
+      this.homeAddress.addressLine3 = ''
+      this.homeAddress.addressLine4 = ''
+      this.homeAddress.postalCode = ''
+      this.mailingAddress.addressLine1 = ''
+      this.mailingAddress.addressLine2 = ''
+      this.mailingAddress.addressLine3 = ''
+      this.mailingAddress.addressLine4 = ''
+      this.mailingAddress.postalCode = ''
```

Comment:
I think you can simplify this by just resetting homeAddress/mailingAddress to an empty object {}.